content/graph/LCT.h

```diff
@@ -0,0 +1,116 @@
+/**
+ * Author: 
+ * Description: link-cut Tree. Supports BBST-like 
+ * augmentations. (Can be used in place of HLD).
+ * Current implementation supports update value at a node,
+ * and query max on a path.
+ * Tested on: http://acm.timus.ru/problem.aspx?num=1553
+ * Status: Passes existing fuzz tests (with function names modified).
```

Comment:
We'll want to remove the old link-cut tree, so feel free to do that and modify the fuzz-test. Will also need to update the test to verify that max works (with max replaced by some non-commutative, non-associative function)

Reply:
Updated the fuzz-test to verify max for now. Will update it later for any new function that we decide on...